fairlearn/grid_search.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+This module implements grid searches for fairness/accuracy tradeoffs. Each
+point in the grid corresponds to a different accuracy/fairness trade-off. The
+dimensionality of the search increases with the number of possible values
+for the protected attribute, so for now only binary protected attributes are
+considered
+"""
+
+import copy
+import numpy as np
+
+class BinaryProtectedAttribute:
+    def __init__(self):
+        # Nothing to do
+        return
+
+    def _generate_p0_p1(self, protected_attribute):
+        """ Function to compute p0 and p1 for the given
+        set of protected_attribute values
+        """
+        unique_labels, counts = np.unique(protected_attribute, return_counts=True)
+        if len(unique_labels) > 2:
+            raise RuntimeError("Protected Attribute is not binary")
+
+        p0 = counts[0] / len(protected_attribute)
+        # It is possible that our data only contain one value for
+        # the protected attribute
+        # We do not consider this an error
+        if len(counts) > 1:
```

Comment:
if we have only ones (label), then counts[0] is the count for 1, not for 0

Reply:
I was just working on something related. This is getting super fiddly when we allow non-numeric labels. Basically, I'm taking
p_0 = n_{A==a_0} / n
but this case will have a_0 == 1, since there's no other value

Reply:
I'm pretty sure everything is consistent now